Auto submodule update: cmake - Merge pull request #80 from swift-nav/jdiamond/load-whole-library-function

swiftnav-svc-jenkins opened the pull request:
Automated PR by Jenkins. If CI has passed successfully, merge away!

**cmake**  3a6f1a72  ->  b12ed266
-  b12ed266  :  Merge pull request #80 from swift-nav/jdiamond/load-whole-library-func
-  9e6a7c10  :  Added script for forcing whole library to be linked


This pull request was created by https://jenkins.ci.swift-nav.com/job/CI%20Infra/job/submodule-update/6082/.

martin-swift commented:
are we overriding the log_ function in those repos? I'm running out of time before I disappear for a week, so we can either let this pull hang out until I can verify it, or maybe someone else can test it?  It "should" work just fine...  There might be some caveats, though.  It seems that link order might be important - from what I understand, when linking against static libs, the linker will resolve the first definition it finds unless you specifically tell it to re-scan all the objects, in which case it should resolve strong symbols over weak ones.  Sounds like maybe I should verify this when I get back.

anth-cole commented:
I'd agree - I don't think this is blocking anything so let's verify when you return - have a good trip!

martin-swift commented:
Thanks!  I thought about it some more - it should still build as a shared lib for LSNP because it will inherit that flag; the only change should be that it won't force other projects (like gnss-converters) to build shared libs.

mbavaro commented on include/swiftnav/logging.h:
```diff
@@ -43,20 +43,17 @@ extern "C" {
  *
  * \{ */
 
-extern void log_(u8 level, const char *msg, ...) __attribute__((weak))
```
I agree, both `extern` and `weak` are inappropriate in the "header" file

mbavaro commented on src/logging.c:
```diff
@@ -25,7 +25,7 @@
  * \param msg Log contents
  */
-void log_(u8 level, const char *msg, ...) {
+__attribute__((weak)) void log_(u8 level, const char *msg, ...) {
```
it's just aesthetics probably, but I have seen this after the `()` in the past

mbavaro commented on src/logging_common.c:
```diff
@@ -27,21 +27,6 @@ const char *level_string[] = {
- * that get overridden by `src/logging.c` and `src/logging_network.cc` for
- * `swiftnav` and `network-swiftnav`.
- */
-void log_(__attribute__((unused)) u8 level,
```
huh.. the weak functions are defined in `logging.c`, well spotted 👍

mbavaro commented:
I checked out this branch within https://github.com/swift-nav/gnss-converters and verified that `cmake ../ -DCMAKE_SYSTEM_NAME=Linux -DBUILD_SHARED_LIBS=OFF` builds correctly a self-contained executable (provided https://github.com/swift-nav/libsbp/pull/592 is used)

martin-swift commented:
@mbavaro - did you get around to trying it with LSNP and piksi fw? @anth-cole - what tests do you want to see before we call this merge-able?

mbavaro commented:
I haven't, I can try now

martin-swift commented:
Yeah, I think there may be more work involved beyond the scope of this pull for integrating into other projects.  @anth-cole - which one of your projects is using this?  what was the motivator for open sourcing it?  I think we should make sure this doesn't break any of the projects that are actually using it and worry about the others later.

mbavaro commented:
also, these two guys are gone:

https://github.com/swift-nav/libswiftnav-private/blob/master/refactor/common/libswiftnav/glonass_phase_biases.h#L56-L60

and I agree with it, but I don't know where to put them now. Compile fails on `/home/snav/lsnp/test_pvt_engine/replay/sbp_obs.cc`

swiftnav-svc-jenkins committed:
cmake - Merge pull request #80 from swift-nav/jdiamond/load-whole-library-function
Added script for forcing whole library to be linked

Triggered-By:	cmake	b12ed26666e0915feec0142464798dd43b3caf31
